core/trino-main/src/main/java/io/trino/sql/planner/plan/TableWriterNode.java

```diff
@@ -573,10 +652,22 @@ public SchemaTableName getSchemaTableName()
             return schemaTableName;
         }
 
+        @JsonProperty
+        public boolean isReportingWrittenBytesSupported()
+        {
+            return reportingWrittenBytesSupported;
+        }
+
         @Override
         public String toString()
         {
             return executeHandle.toString();
         }
+
+        @Override
+        public boolean supportsReportingWrittenBytes(Metadata metadata, Session session)
+        {
+            return sourceHandle.map(tableHandle -> metadata.supportsReportingWrittenBytes(session, tableHandle)).orElse(reportingWrittenBytesSupported);
```

Comment:
why not simply `return reportingWrittenBytesSupported`?

Reply:
It is for case when `sourceHandle` will be changed after optimizations (then, it will be set to non-empty value). It is a kind of tradeof in some sense - this what we agreed together with @losipiuk